test/integration_test.py

```diff
@@ -2034,25 +2038,66 @@ def test_response_security_headers(self):
             '/oauth2_redirect': {'Cache-Control': 'no-store'},
             '/health/basic': {'Cache-Control': 'no-store'}
         }
+        # https://www.w3.org/TR/CSP2/#policy-syntax
+        directive_re = re.compile(r'[ \t]*([a-zA-Z0-9-]+)'
+                                  # Space, tab and any visible character
+                                  # (0x21-0xFE) except for comma (0x2C) or
+                                  # semicolon (0x3B).
+                                  r'(?:[ \t]([ \t\x21-\x2B\x2D-\x3A\x3C-\xFE]*))?')
+        nonce_re = re.compile(r"'nonce-([a-zA-Z0-9+/]{43})'")
+        expected_expressions = {
+            sq('self'),
+            sq('none'),
+            'data:'
+        }
+        expected_directives = {
+            'default-src',
+            'img-src',
+            'script-src',
+            'style-src',
+            'frame-ancestors'
+        }
```

Comment:
This is a regression to the point where we duplicated the the expected headers in assertions. The ONLY unpredictable part is the nonce so the previous approach of updating the expected headers with that nonce value is much less brittle, as long as the parsing of the CSP and the validation of the nonce's syntax doesn't let invalid or duplicate nonces through.

Reply:
Note that with the refactoring of the parsing & validation of the CSP into a separate method, I had to remove the CSP from the expected headers due to no longer having the nonce value available to generate the expected CSP in the test.

Reply:
> Note that with the refactoring of the parsing & validation of the CSP into a separate method, I had to remove the CSP from the expected headers due to no longer having the nonce value available to generate the expected CSP in the test.

I don't understand. Just have the parsing/validation method return the expected header entry and inject the return value into the expected header dictionary. Please raise in PL if this is not clear.